src/tar.toit

```diff
@@ -12,27 +12,36 @@ Writes the given files into the writer in tar file format.
 class Tar:
   writer_ ::= ?
 
+  /**
+  Creates a new Tar archiver that writes to the given $writer.
+
+  The $writer should be an $io.Writer, but for compatibility reasons, it can also be an
+    "old-style" writer. The old-style writer is deprecated and will be removed in the future.
+  */
   constructor writer:
-    writer_ = Writer writer
+    if writer is io.Writer:
+      writer_ = writer
+    else:
+      writer_ = io.Writer.adapt writer
 
   /**
   Adds a new "file" to the generated tar-archive.
 
   This function sets all file attributes to some default values. For example, the
     modification date is set to 0 (epoch time).
   */
-  add file_name/string content -> none:
+  add file_name/string content/io.Data -> none:
     add_ file_name content --type=normal_
 
   /**
-  Closes the tar stream, and invokes `close_write` on the stored writer if $close_writer is true.
+  Closes the tar stream, and invokes 'close' if $close_writer is true.
```

Comment:
Invokes close on what? I hope it is not infinite recursion :)

Reply:
done.